feat: improve chatbot integration + css fixes

shanegrouber opened the pull request:
<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

## Release Notes

- **New Features**
  - Introduced a new configuration option for the Botpress client ID, enhancing the environment schema.
  - Enhanced the Chatbot component to manage client instances dynamically based on user interactions.
  - Added a new `chatbot` feature in the customer schema to control chatbot availability.

- **Improvements**
  - Updated the Root page to better integrate the chatbot feature within an iframe, allowing for improved visibility and configuration control.

- **Schema Updates**
  - Removed the `isChatbotEnabled` property from the WorkflowConfigSchema and ConfigSchema, streamlining the schema structure.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

shanegrouber committed:
feat(chatbot): enhance chatbot functionality with case data handling

- Add client as a prop to Chatbot component
- Implement sending current case data to chatbot
- Improve environment variable usage for client ID

(Your code is like a bad magician—now you see the client, now you don’t)

shanegrouber committed:
feat(chatbot): wrap chatbot in iframe component

- Import RenderChildrenInIFrame to contain the Chatbot
- Adjust resizing behavior based on chatbot visibility

(It's about time that chatbot got some boundaries; it was getting a bit too cozy)

shanegrouber committed:
feat(chatbot): add configurable client ID for chatbot integration

- Introduce VITE_BOTPRESS_CLIENT_ID to environment schema
- Update Chatbot component to utilize dynamic client ID
- Adjust customer features to include chatbot configuration

(If your chatbot was any less reliable, it would be giving out horoscopes instead of responses)

shanegrouber committed:
chore(chatbot): remove unnecessary chatbot enable flags

- Remove redundant isChatbotEnabled flags from schemas
- Clean up caseId assignment placement in chatbot component

(Your code is so clean, it's practically begging for a dust bunny)

tomer-shvadron commented on apps/backoffice-v2/src/domains/chat/chatbot-opengpt.tsx:
```diff
@@ -1,38 +1,93 @@
+  }, [session, client, setClient, sendCaseData, pathname, chatbotClientId]);
+
+  useEffect(() => {
+    console.log('pathname changed to: ', pathname);
```
Please remove all instances `console.log` in the PR

tomer-shvadron commented on apps/backoffice-v2/src/domains/chat/chatbot-opengpt.tsx:
```diff
@@ -1,38 +1,93 @@
+
+    const caseId = pathname.split('/')[pathname.split('/').length - 1];
+
+    if (caseId) void sendCaseData(caseId);
```
```suggestion
    if (caseId) {
      void sendCaseData(caseId);
    }
```

tomer-shvadron commented on apps/backoffice-v2/src/domains/chat/chatbot-opengpt.tsx:
```diff
@@ -1,38 +1,93 @@
+      setTimeout(() => {
+        const caseId = pathname.split('/')[pathname.split('/').length - 1];
+        void sendCaseData(caseId || '', newClient);
+      }, 500);
```
Try setting the timeout to 0 and see if it still works

tomer-shvadron commented on apps/backoffice-v2/src/domains/chat/chatbot-opengpt.tsx:
```diff
@@ -1,38 +1,93 @@
 const Chatbot = ({
   isWebchatOpen,
   toggleIsWebchatOpen,
+  client,
```
Check if there's any reason not to move `client` and `setClient` initialization into the Chatbot component rather then getting it as a parameter

tomer-shvadron commented on apps/backoffice-v2/src/domains/chat/chatbot-opengpt.tsx:
```diff
@@ -1,38 +1,93 @@
-  const client = getClient({ clientId });
   const { data: session } = useAuthenticatedUserQuery();
+  const { data: currentCase } = useCurrentCaseQuery();
+  const { pathname } = useLocation();
```
Change the use of `useLocation` and `pathname` with `const { entityId: workflowId } = useParams();`

tomer-shvadron commented on apps/backoffice-v2/src/domains/chat/chatbot-opengpt.tsx:
```diff
@@ -1,38 +1,93 @@
+    async (caseId: string, newClient?: WebchatClient) => {
+      if (!currentCase) return;
+
+      const clientToUse = newClient || client;
```
You could remove this line if you'd initialize `newClient` in line 32 like so:
`async (caseId: string, newClient?: WebchatClient = client) => {`
And then you can just use `newClient`.

Also, `newClient` is a bad name. new? old? Try finding a better name

tomer-shvadron commented on apps/backoffice-v2/src/domains/chat/chatbot-opengpt.tsx:
```diff
@@ -1,38 +1,93 @@
+        };
+        await clientToUse?.sendEvent({
+          type: 'case-data',
+          data: currentCaseData,
```
Try and see if `currentCaseData` const is needed and if you can just use `data: currentCase.context` in line 44

tomer-shvadron commented on apps/backoffice-v2/src/domains/chat/chatbot-opengpt.tsx:
```diff
@@ -1,38 +1,93 @@
+    const newClient = getClient({ clientId: chatbotClientId });
+    setClient(newClient);
+
+    // newClient.on('*', (ev: any) => {
```
Remove comments

tomer-shvadron reviewed (approved):
<img src="https://media1.giphy.com/media/67WcBZ0crjuKZ4aTDt/giphy.gif"/>

shanegrouber committed:
refactor(chatbot): streamline chatbot client management

- Remove unused WebchatClient state from ChatbotLayout
- Change variable names for clarity and consistency
- Update useEffect hooks to handle new parameter definitions

(With all these changes, I half-expect the chatbot to start giving better life advice)

tomer-shvadron committed:
Merge branch 'dev' into BAL-3040

shanegrouber committed:
refactor(customer): rename 'isEnabled' to 'enabled' for clarity

- Update CustomerSchema to use 'enabled' instead of 'isEnabled'
- Adjust conditional check in ChatbotLayout for consistency

(your naming conventions are so confusing, they should come with a user manual)
